### ISSUE TITLE
`QCheck.char` doesn't shrink (but `QCheck2.Gen.char` does)

### ISSUE THREAD

**Issue by jmid** [c-cube/qcheck#166]
I was surprised to discover this - illustrated by test `char_is_never_abcdef`.
`QCheck` reports a `'d'` counterexample, while `QCheck2` shrinks it to `a`:
```
Test char is never produces 'abcdef' failed (0 shrink steps): |	Test char is never produces 'abcdef' failed (1 shrink steps):

'd'							      |	'a'
```
This is a defect in `QCheck.char` IMO - users would expect it to shrink.

_Originally posted by @jmid in https://github.com/c-cube/qcheck/issues/153#issuecomment-898715367_

**Comment by jmid** [comment]
We even have a `QCheck.Shrink.char` shrinker:
```ocaml
let char c yield =
    if Char.code c > 0 then yield (Char.chr (Char.code c-1))
```
It just isn't used by `QCheck.char`! 🤔 As a first version, we could consider redefining the `char arbitrary` as:
```ocaml
let char = make_scalar ~print:(sprintf "%C") ~shrink:Shrink.char Gen.char
```
Potentially it could be refined to shrink towards `'a'` like QCheck2.

**Comment by jmid** [comment]
Fixed by #247
